### PR TITLE
Initial support of Glue catalog table and Jobs APIs

### PR DESCRIPTION
**Pull request by arthurnovak**
Please review @motobob @nalundgaard

### CODE REVIEW

File: src/erlcloud_glue.erl [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+               {"content-type", "application/x-amz-json-1.1"}],
+    Region = erlcloud_aws:aws_region_from_host(Host),
+    erlcloud_aws:sign_v4_headers(Config, Headers, Body, Region, "glue").
+
+get_url(#aws_config{glue_scheme = Scheme,
+                    glue_host   = Host,
+                    glue_port   = Port}) ->
+    Scheme ++ Host ++ ":" ++ integer_to_list(Port).
+
+get_retry_fun() ->
```
Comment (motobob) [inline src/erlcloud_glue.erl]: why not use the fun directly ?

File: src/erlcloud_glue.erl [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+    Request1 = update_catalog_id(CatalogId, Request0),
+    case request(Config, "CreateTable", Request1) of
+        {ok, _} -> ok;
+        Error   -> Error
+    end.
+
+%% @doc
+%% Glue API:
+%% https://docs.aws.amazon.com/glue/latest/webapi/API_DeleteJob.html
+%%
```
Comment (nalundgaard) [inline src/erlcloud_glue.erl]: Any reason for not including a sample invocation here?

Comment (arthurnovak) [reply src/erlcloud_glue.erl]: i don't think it makes sense to document any obvious fun arguments and responses everywhere. We have spec for that, which looks descriptive in current case: jobname binary as input, ok as output:
```erlang
-spec delete_job(binary()) -> ok | {error, any()}.
delete_job(JobName) ->
    delete_job(JobName, default_config()).
```
The main problem with docs: they are forgotten to update very often
In other cases, where we have some not obvious arguments like `TableInput` as map, I've added docs

Comment (nalundgaard) [reply src/erlcloud_glue.erl]: 👍

**Pull request merged**